tests/E2E-Puppeteer/test/linkchecker.js

```diff
@@ -0,0 +1,143 @@
+    if (status.startsWith('4') || status.startsWith('5')) {
+      console.log(' * Failed response in ' + office + ' from ' + cur_href + ': ' + url + ' (' + status + ')');
+      object[office].failed.push({page: cur_href, URL: url, status: status});
+    } else {
+      responses.push({url: url, status: status});
+    }
+  });
+};
+
+const checkStatusUrls = async (page, hrefs) => {
+  //check urls
+  for (const href of hrefs) {
+    cur_href = href;
+    object[office].passed.push({url: href, date: new Date().getTime(), responses: responses});
+
+    if (!href.includes('mailto:') && href.includes(URL_FO)) {
+      await page.goto(href, {waitUntil: 'domcontentloaded'});
+    }
+    responses = [];
+  }
+  object[office].totalRequests = number_requests[office];
+};
+
+const login_BO = async (page) => {
+  await page.type('#email', EMAIL);
+  await page.type('#passwd', PASSWD);
+  await page.click('#submit_login');
+  await page.waitForNavigation({waitUntil: 'domcontentloaded'});
+};
+
+const run = async () => {
+  let browser = await puppeteer.launch({
+    headless: HEADLESS,
+    args: ['--no-sandbox']
+  });
+  let page = await browser.newPage();
+  await interceptRequestAndResponse(page, office);
+  console.time('Check urls ');
+
+  //Start testing BO
+  console.log('Begin testing BO');
+  await page.goto(URL_BO, {waitUntil: 'networkidle0'});
+  await login_BO(page);
+  let urlList = await getAllUrls(page, selector_BO);
+  console.log(' - ' + urlList.length + ' URL to crawl');
+  office = 'BO';
+  object.BO.totalLinks = urlList.length;
+  await checkStatusUrls(page, urlList);
+  console.log('End testing BO');
+  // End testing BO
+
+  // Start testing FO
+  console.log('Begin testing FO');
+  await page.goto(URL_FO, {waitUntil: 'networkidle0'});
+  urlList = await getAllUrls(page, selector_FO);
+  console.log(' - ' + urlList.length + ' URL to crawl');
+  office = 'FO';
+  object.FO.totalLinks = urlList.length;
+  await checkStatusUrls(page, urlList);
+  console.log('End testing FO');
+  // End testing FO
+
+  browser.close();
+  console.timeEnd('Check urls ');
+
+  object.stats.end = new Date().getTime();
+
+  const filename = 'report_' + new Date().toJSON().slice(0, 10) + '.json';
+  fs.writeFile(report_path + filename, JSON.stringify(object), function (err) {
+    if (err) {
+      return console.log(err);
+    }
+    console.log("File " + report_path + filename + " saved!");
+  });
+
+  //is there any error when crawling ?
+  if (object.BO.failed.length > 0 || object.FO.failed.length > 0) {
+    let error_message = '';
+    if (object.BO.failed.length > 0) {
+      error_message += ' - ' + object.BO.failed.length + " errors in BO";
+    }
+    if (object.FO.failed.length > 0) {
+      error_message += ' - ' + object.FO.failed.length + " errors in FO";
+    }
+    throw ('Errors caught : ' + error_message);
+  }
+};
+
+run()
+  .then(() => {
+    console.log("--------the end--------")
+  }).catch(e => console.log(`error: ${e}`));
```

Comment:
Should be better to use something like https://www.npmjs.com/package/check-broken-links to retrieve report data.

Reply:
Actually, this script does a little more than just report 500: it also reports failing requests, like CSS/JS/images 404'ing.
We will try to use chai and maybe mocha too, just to be able to generate a report compatible with our QA board.